Mathlib/Topology/ContinuousMap/StarOrdered.lean

```diff
@@ -3,63 +3,50 @@ Copyright (c) 2024 Jireh Loreaux. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jireh Loreaux
 -/
-import Mathlib.Analysis.Complex.Basic
-import Mathlib.Data.Real.StarOrdered
+import Mathlib.Algebra.Order.Star.Basic
 import Mathlib.Topology.ContinuousMap.Algebra
 import Mathlib.Topology.ContinuousMap.ContinuousMapZero
 
-/-! # Continuous functions as a star-ordered ring -/
+/-! # Continuous functions as a star-ordered ring
 
-open scoped NNReal
+The type class `ContinuousSqrt` gives a sufficient condition on `R` to make `C(α, R)`
+and `C(α, R)₀` into a `StarOrderedRing` for any topological space `α`, thereby providing a means
+by which we can ensure `C(α, R)` has this property. This condiiton is satisfied
+by `ℝ≥0`, `ℝ`, and `ℂ`, and the instances can be found in the file
+`Topology.ContinuousMap.ContinuousSqrt`.
+-/
+
+/-- A type class encoding the property that there is a continuous square root function on
+nonnegative elements. This holds for `ℝ≥0`, `ℝ` and `ℂ` (as well as any C⋆-algebra), and this
+allows us to derive an instance of `StarOrderedRing C(α, R)` under appropriate hypotheses.
+In order for this to work on `ℝ≥0`, we actually must force our square root function to be defined
+on and well-behaved for pairs `x : R × R` with `x.1 ≤ x.2`. -/
+class ContinuousSqrt (R : Type*) [LE R] [NonUnitalSemiring R] [TopologicalSpace R] where
+  /-- `sqrt (a, b)` returns a value `s` such that `b = a + s * s` when `a ≤ b`. -/
```

Comment:
This spelling avoiding subtraction feels a little strange to me; do we lose any interesting generality by using `OrderedSub` instead?

Reply:
The issue is that it proliferates the type classes I would need to assume on the scalar ring for the continuous functional calculus, which would be annoying.

Reply:
Could you add a helper function that assumes `OrderedSub`, and use it to prove `ContinuousSqrt ℝ≥0`? I know it doesn't make the proof shorter, but it leaves somewhere for you to put a comment explaining why actually defining things that way is annoying.

Reply:
@eric-wieser I decided instead to simply add a bit to the module documentation explaining the situation and reasoning.